.travis.yml

```diff
@@ -206,39 +207,55 @@ base_osx_config: &base_osx_config
     brew:
       packages:
       - openssl
-  before_install:
-    - curl -L https://github.com/stedolan/jq/releases/download/jq-1.5/jq-osx-amd64 -o /usr/local/bin/jq
-    - chmod 755 /usr/local/bin/jq
-    - ./build-support/bin/install_aws_cli_for_ci.sh
-    - ./build-support/bin/install_python_for_ci.sh "${PYENV_PY36_VERSION}"
 
 py27_osx_config: &py27_osx_config
   <<: *base_osx_config
+  # NB: We ensure sane python2 and python3 interpreters are available on the PATH for scripting
+  # against.
   env:
     - &py27_osx_config_env >
-      PATH="/usr/local/opt/openssl/bin:$PATH"
+      PATH="/usr/local/opt/openssl/bin:${PATH}"
       LDFLAGS="-L/usr/local/opt/openssl/lib"
       CPPFLAGS="-I/usr/local/opt/openssl/include"
+      PATH="${PYENV_ROOT}/versions/${PYENV_PY27_VERSION}/bin:${PATH}"
+      PATH="${PYENV_ROOT}/versions/${PYENV_PY36_VERSION}/bin:${PATH}"
+  before_install:
+    - curl -L https://github.com/stedolan/jq/releases/download/jq-1.5/jq-osx-amd64 -o /usr/local/bin/jq
+    - chmod 755 /usr/local/bin/jq
+    - ./build-support/bin/install_aws_cli_for_ci.sh
+    - ./build-support/bin/install_python_for_ci.sh "${PYENV_PY27_VERSION}"
+    - ./build-support/bin/install_python_for_ci.sh "${PYENV_PY36_VERSION}"
 
 py36_osx_config: &py36_osx_config
   <<: *base_osx_config
   env:
     - &py36_osx_config_env >
-      PATH="/usr/local/opt/openssl/bin:$PATH"
+      PATH="/usr/local/opt/openssl/bin:${PATH}"
       LDFLAGS="-L/usr/local/opt/openssl/lib"
       CPPFLAGS="-I/usr/local/opt/openssl/include"
+      PATH="${PYENV_ROOT}/versions/${PYENV_PY27_VERSION}/bin:${PATH}"
+      PATH="${PYENV_ROOT}/versions/${PYENV_PY36_VERSION}/bin:${PATH}"
+  before_install:
+    - curl -L https://github.com/stedolan/jq/releases/download/jq-1.5/jq-osx-amd64 -o /usr/local/bin/jq
+    - chmod 755 /usr/local/bin/jq
+    - ./build-support/bin/install_aws_cli_for_ci.sh
+    - ./build-support/bin/install_python_for_ci.sh "${PYENV_PY27_VERSION}"
+    - ./build-support/bin/install_python_for_ci.sh "${PYENV_PY36_VERSION}"
```

Comment:
The script takes multiple args. Can simplify to `./build-support/bin/install_python_for_ci.sh "${PYENV_PY27_VERSION}"  "${PYENV_PY36_VERSION}"`.

Beyond less lines, it's important to do in one call so that the script's call to `pyenv global` is `pyenv global 2.7 3.6`, otherwise the second script will override the first call to `pyenv global`.

Reply:
No-no-no. 1st, seperate lines is more readable. Second, I've killed all the shims/global business.

Reply:
Eh, maybe not so readable. At one point I had installed all 3 on all osx shards. That said - still nuked the global bit from the script to prove the point and stay well away from shims shenanigans. I'll see what falls out in CI as I work this.

Reply:
Disagree on the readability - I find one line more readable because they’re highly related so good to keep together. But fine with me if you want to keep the change. 

Agreed on avoiding shims. That simplifies everything and is a great improvement. I remember I tried that with the Docker shards and couldn’t get it to work, so gave up and went back to shims.

Reply:
Yeah, one unenforced thing has been 100 cols on all files and a bunch of violations have crept in. I think even that may be ok now with just 2 versions instead of 3.

Reply:
Fixed.